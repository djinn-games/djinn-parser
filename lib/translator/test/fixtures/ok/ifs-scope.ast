{
  "type": "Program",
  "name": {
    "type": "Identifier",
    "name": "ifs_scope",
    "line": 1
  },
  "body": {
    "type": "Body",
    "sentences": [
      {
        "type": "ExpressionSentence",
        "expression": {
          "type": "VarDeclaration",
          "dataType": "bool",
          "id": {
            "type": "Identifier",
            "name": "x",
            "line": 3
          },
          "init": {
            "type": "Literal",
            "line": 3,
            "dataType": "bool",
            "value": false
          },
          "line": 3
        },
        "line": 3
      },
      {
        "type": "IfSentence",
        "if": {
          "condition": {
            "type": "VarDeclaration",
            "dataType": "bool",
            "id": {
              "type": "Identifier",
              "name": "x",
              "line": 4
            },
            "init": {
              "type": "Literal",
              "line": 4,
              "dataType": "bool",
              "value": false
            },
            "line": 4
          },
          "consequent": [
            {
              "type": "ExpressionSentence",
              "expression": {
                "type": "VarDeclaration",
                "dataType": "int",
                "id": {
                  "type": "Identifier",
                  "name": "y",
                  "line": 5
                },
                "init": {
                  "type": "Literal",
                  "line": 5,
                  "dataType": "int",
                  "value": 1
                },
                "line": 5
              },
              "line": 5
            }
          ],
          "alternates": [
            [
              {
                "type": "IfSentence",
                "if": {
                  "condition": {
                    "type": "VarDeclaration",
                    "dataType": "bool",
                    "id": {
                      "type": "Identifier",
                      "name": "x",
                      "line": 6
                    },
                    "init": {
                      "type": "Literal",
                      "line": 6,
                      "dataType": "bool",
                      "value": true
                    },
                    "line": 6
                  },
                  "consequent": [
                    {
                      "type": "ExpressionSentence",
                      "expression": {
                        "type": "VarDeclaration",
                        "dataType": "int",
                        "id": {
                          "type": "Identifier",
                          "name": "y",
                          "line": 7
                        },
                        "init": {
                          "type": "Literal",
                          "line": 7,
                          "dataType": "int",
                          "value": 2
                        },
                        "line": 7
                      },
                      "line": 7
                    }
                  ],
                  "alternates": []
                },
                "line": 6
              }
            ]
          ],
          "line": 4
        },
        "line": 4
      }
    ],
    "line": 2
  },
  "line": 1
}