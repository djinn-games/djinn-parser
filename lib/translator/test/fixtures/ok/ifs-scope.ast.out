{
  "type": "Program",
  "body": [
    {
      "type": "ExpressionStatement",
      "expression": {
        "type": "AssignmentExpression",
        "operator": "=",
        "left": {
          "type": "MemberExpression",
          "object": {
            "type": "MemberExpression",
            "object": {
              "type": "Identifier",
              "name": "DJINN"
            },
            "property": {
              "type": "Identifier",
              "name": "__scope"
            }
          },
          "property": {
            "type": "Literal",
            "value": "_x",
            "raw": "\"_x\""
          },
          "computed": true
        },
        "right": {
          "type": "Literal",
          "value": false,
          "raw": "false"
        }
      }
    },
    {
      "type": "IfStatement",
      "test": {
        "type": "AssignmentExpression",
        "operator": "=",
        "left": {
          "type": "MemberExpression",
          "object": {
            "type": "MemberExpression",
            "object": {
              "type": "Identifier",
              "name": "DJINN"
            },
            "property": {
              "type": "Identifier",
              "name": "__scope"
            }
          },
          "property": {
            "type": "Literal",
            "value": "__x",
            "raw": "\"__x\""
          },
          "computed": true
        },
        "right": {
          "type": "Literal",
          "value": false,
          "raw": "false"
        }
      },
      "consequent": {
        "type": "BlockStatement",
        "body": [
          {
            "type": "ExpressionStatement",
            "expression": {
              "type": "AssignmentExpression",
              "operator": "=",
              "left": {
                "type": "MemberExpression",
                "object": {
                  "type": "MemberExpression",
                  "object": {
                    "type": "Identifier",
                    "name": "DJINN"
                  },
                  "property": {
                    "type": "Identifier",
                    "name": "__scope"
                  }
                },
                "property": {
                  "type": "Literal",
                  "value": "__y",
                  "raw": "\"__y\""
                },
                "computed": true
              },
              "right": {
                "type": "Literal",
                "value": 1,
                "raw": "1"
              }
            }
          }
        ]
      },
      "alternate": {
        "type": "IfStatement",
        "test": {
          "type": "AssignmentExpression",
          "operator": "=",
          "left": {
            "type": "MemberExpression",
            "object": {
              "type": "MemberExpression",
              "object": {
                "type": "Identifier",
                "name": "DJINN"
              },
              "property": {
                "type": "Identifier",
                "name": "__scope"
              }
            },
            "property": {
              "type": "Literal",
              "value": "__x",
              "raw": "\"__x\""
            },
            "computed": true
          },
          "right": {
            "type": "Literal",
            "value": true,
            "raw": "true"
          }
        },
        "consequent": {
          "type": "BlockStatement",
          "body": [
            {
              "type": "ExpressionStatement",
              "expression": {
                "type": "AssignmentExpression",
                "operator": "=",
                "left": {
                  "type": "MemberExpression",
                  "object": {
                    "type": "MemberExpression",
                    "object": {
                      "type": "Identifier",
                      "name": "DJINN"
                    },
                    "property": {
                      "type": "Identifier",
                      "name": "__scope"
                    }
                  },
                  "property": {
                    "type": "Literal",
                    "value": "__y",
                    "raw": "\"__y\""
                  },
                  "computed": true
                },
                "right": {
                  "type": "Literal",
                  "value": 2,
                  "raw": "2"
                }
              }
            }
          ]
        },
        "alternate": {
          "type": "BlockStatement",
          "body": []
        }
      }
    }
  ]
}
